pkg/authorization/authorizer/attributes.go

```diff
@@ -85,7 +99,15 @@ func verbMatches(a Action, verbs sets.String) bool {
 }
 
 func resourceMatches(a Action, allowedResourceTypes sets.String) bool {
-	return allowedResourceTypes.Has(authorizationapi.ResourceAll) || allowedResourceTypes.Has(strings.ToLower(a.GetResource()))
+	if allowedResourceTypes.Has(authorizationapi.ResourceAll) {
+		return true
+	}
+
+	if len(a.GetSubresource()) == 0 {
+		return allowedResourceTypes.Has(strings.ToLower(a.GetResource()))
+	}
+
+	return allowedResourceTypes.Has(strings.ToLower(a.GetResource() + "/" + a.GetSubresource()))
```

Comment:
> this is going to get called a lot... should find a way to avoid allocating on every call

I don't see this as a blocker for merge.  The interfaces need to be right for the rebase, the call path optimization only needs to be done by release.